avocado/core/runner.py

```diff
@@ -466,9 +466,9 @@ def run_suite(self, test_suite, mux, timeout=0, replay_map=None,
             TEST_LOG.error('Job interrupted by ctrl+c.')
             summary.add('INTERRUPTED')
 
-        self.result.end_tests()
-        self.job.funcatexit.run()
         if self.job.sysinfo is not None:
             self.job.sysinfo.end_job_hook()
+        self.result.end_tests()
+        self.job.funcatexit.run()
```

Comment:
Actually the `end_tests` method means all tests finished, but based on the usage, it's more like `end_job` (it's the last interface between results and job). So if I think about this more, probably even the `job.funcatexit` should be executed before we produce the `end_tests` (which in the ideal world would be `end_job`, but it's too late to change it now).